bec2format/bf3file.py

```diff
@@ -1,35 +1,21 @@
-from binascii import unhexlify
 from collections import namedtuple
-from collections.abc import Iterator
-from re import sub
-from typing import Any, TextIO
-
-from bec2format.bytes_reader import BytesReader
-from bec2format.configid import ConfDict, ConfigId
-from bec2format.crypto import create_AES128, pad
-from bec2format.error import (
+from typing import Any, Iterable, Optional, TextIO
+
+from .bytes_reader import BytesReader, hex2bin
+from .configid import ConfDict, ConfigId
+from .crypto import create_AES128, pad
+from .error import (
     Bf3FileFormatError,
     MissingDeviceSettingsNameError,
     MissingProjectSettingsNameError,
     UnsupportedBf2InstrError,
     UnsupportedLegacyFirmwareError,
     UnsupportedTagTypeError,
 )
-from bec2format.hwcids import HWCID_MAP, REV_HWCID_MAP
-
-_space_chars = " \t\r\n,-/:"
-_hex2bin_translate_table = str.maketrans(_space_chars, " " * len(_space_chars))
-
-
-def hex2bin(hex_value: str) -> bytes:
-    clean_hex_str = hex_value.translate(_hex2bin_translate_table)
-    clean_hex_str = sub(r"\s", r"", clean_hex_str)
-    if len(clean_hex_str) % 2 == 1:
-        clean_hex_str = clean_hex_str[:-1] + "0" + clean_hex_str[-1]
-    return unhexlify(clean_hex_str)
```

Comment:
Ich würde `hex2bin()` aus mehreren Gründen in diesem Modul belassen:
- in BytesReader gehört es auch nicht (hat nichts mit dem einlesen eines Streams zu tun)
- Es handelt sich um eine spezielle implementierung zum einlesen von BF2 files (die im gegensatz zu `bytes.from_hex()` auch ':' und andere Zeichen verarbeiten kann)
- Es wurde nach meinem Verständnis nur wegen `.crypto` in ein zentrales Modul verschoben. Für `crypto` sollte aber die Standard-Funktion `unhexlify()` ausreichen. Dann entfällt auch die (nicht notwendige) Abhängigkeit zu `.bytes_reader`

Reply:
`bytes.fromhex()` ist anders als zuerst behauptet in Micropython 1.20 implementiert (unterstützt lediglich keine Leer und Sonderzeichen im übergebenen String). Der betroffene Commit ist entfernt, d.h. `hex2bin()` ist wieder im `bf3file` Modul und wird ausschließlich dort benutzt. An allen anderen Stellen wird `bytes.fromhex()` mit entfernten Leer- und Sonderzeichen im Argument benutzt.